lib/app/features/feed/providers/image_aspect_ratio_cache.dart

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/services/media_service/aspect_ratio.dart';
+
+final imageAspectRatioCacheProvider = Provider<ImageAspectRatioCache>((ref) {
```

Comment:
It will create a singleton in terms of Riverpod. Need to use Provider.autoDispose or replace with the generated provider to align with one style in the project